src/components/toggle-fullscreen.tsx

```diff
@@ -0,0 +1,28 @@
+import { useState } from 'react'
+import { Maximize, Minimize } from 'lucide-react'
+import { Button } from '@/components/ui/button'
+
+export function ToggleFullscreen() {
+  const [isFullscreen, setIsFullscreen] = useState(false)
+  const toggleFullscreen = () => {
+    if (!document.fullscreenElement) {
+      document.documentElement.requestFullscreen()
+      setIsFullscreen(true)
+    } else {
+      if (document.exitFullscreen) {
+        document.exitFullscreen()
+        setIsFullscreen(false)
+      }
+    }
+  }
+
+  return (
+    <Button variant='ghost' size='icon' onClick={toggleFullscreen}>
+      {isFullscreen ? (
+        <Minimize className='h-5 w-5' />
+      ) : (
+        <Maximize className='h-5 w-5' />
+      )}
```

Comment:
This logic does not work if you go to full-screen mode, then navigate between pages.